tool/grind.dart

```diff
@@ -5,8 +5,28 @@ import 'dart:io';
 
 import 'package:grinder/grinder.dart';
 
-main(args) => grind(args);
+import 'package:unscripted/unscripted.dart';
 
+main(args) {
+
+  addTask(new GrinderTask(
+      'bump',
+      taskFunction: () {
+        var releaseType = context.invocation.positionals.first;
```

Comment:
I don't anticipate many folks actually needing to use the imperative API introduced in this PR, once I add the follow up annotation-based API which will look something like:

```dart
@Task('bump the version')
bump(
    @Positional(help: 'Which type of release to do', allowed: ['release', 'build', 'patch', 'minor', 'major'])
    String releaseType,
    {bool pre: false
     String preId}) {
  // ...
}
```

which is about as terse as it could possibly get.

The name `invocation` is to match `TaskInvocation`.  I considered `TaskArgs`, which would match up with `args`, but that doesn't really capture the fact that it contains the task name as well.  And "invocation" is already a familiar concept for anyone who has used `dart:mirrors`.